source/MaterialXCore/Interface.cpp

```diff
@@ -13,8 +13,104 @@
 namespace MaterialX
 {
 
+namespace
+{
+    bool validSourceSwizzlePattern(const string &type, const string &channels)
+    {
+        // Check to see if the channels match those supported by the provided type
+        std::set<char> supportedChannels;
+        if (type == TypedValue<float>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'r', 'x' };
+        }
+        else if (type == TypedValue<Color2>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'r', 'a' };
+        }
+        else if (type == TypedValue<Color3>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'r', 'g', 'b' };
+        }
+        else if (type == TypedValue<Color4>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'r', 'g', 'b', 'a' };
+        }
+        else if (type == TypedValue<Vector2>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'x', 'y' };
+        }
+        else if (type == TypedValue<Vector3>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'x', 'y', 'z' };
+        }
+        else if (type == TypedValue<Vector4>::TYPE)
+        {
+            supportedChannels = { '0', '1', 'x', 'y', 'z', 'w' };
+        }
+        else
+        {
+            return false;
+        }
+
+        for (const char& channel : channels)
+        {
+            if (supportedChannels.count(channel) == 0)
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    bool validDestinationSwizzlePattern(const string &type, const string &channels)
+    {
+        // Check to see if the number of channels match the provided type's required number of channels
+        size_t supportedChannelSize = 0;
```

Comment:
Same here with maybe using a map instead, from type string to channels size.

This would also be nicely handled later if we had the `TypeDesc` class in core, just using `TypeDesc::getSize()`.

Reply:
I'll look at this. Must be a better way than using a string to figure out size of a type.